dev/tests/api-functional/testsuite/Magento/GraphQl/Quote/AddSimpleProductToCartCusOptionTest.php

```diff
@@ -0,0 +1,341 @@
+        $checkWithResponse = $selectedDetails["checkWithResponse"];
+        $maskedQuoteId = $quoteIdToMaskedId->execute((int)$quote->getId());
+        $query = $this->prepareAddProductRequestQuery($maskedQuoteId, $sku, $selectedJsonString);
+        $response = $this->graphQlQuery($query);
+        self::assertArrayHasKey("addSimpleProductsToCart", $response);
+        $resOptDetails = $this->processResponseForCustomOption($response);
+        self::assertEquals($checkWithResponse, $resOptDetails, "", 0.0, 10, true);
+    }
+
+    /**
+     * @param \Magento\Catalog\Model\Product $product
+     * @param array $selectedTypes
+     * @return array
+     */
+    public function getSelectedOptionDetails(\Magento\Catalog\Model\Product $product, array $selectedTypes): array
+    {
+        $optionIds = $this->getOptionDetails($product, $selectedTypes);
+        $checkWithResponse = $selectedOption = [];
+        foreach ($selectedTypes as $type => $selectCount) {
+            foreach ($optionIds[$type] as $optId => $optV) {
+                if (is_array($optV)) {
+                    $count = 0;
+                    foreach ($optV as $optVId => $optString) {
+                        if ($count < $selectCount) {
+                            $selectedOption[] = $optString;
+                            $checkWithResponse[$optId][] = (string)$optVId;
+                            $count++;
+                        }
+                    }
+                } else {
+                    $selectedOption[] = $optV;
+                    $checkWithResponse[$optId][] = $selectedTypes[$type];
+                }
+            }
+        }
+        $jsonStr = implode(",", $selectedOption);
+        if (count($selectedOption) > 0) {
+            $jsonStr = "[" . $jsonStr . "]";
+        }
+        return [
+           "jsonString" => $jsonStr,
+           "checkWithResponse" => $checkWithResponse
+        ];
+    }
+
+    /**
+     * @param $response
+     * @return array
+     */
+    public function processResponseForCustomOption($response): array
+    {
+        $resOptDetails = [];
+        if (isset($response["addSimpleProductsToCart"]["cart"]["items"])) {
+            foreach ($response["addSimpleProductsToCart"]["cart"]["items"] as $itemCustomOption) {
+                if (isset($itemCustomOption["customizable_options"])) {
+                    foreach ($itemCustomOption["customizable_options"] as $option) {
+                        foreach ($option["values"] as $optionV) {
+                            $resOptDetails[$option["id"]][] = $optionV["value"];
+                        }
+                    }
+                }
+            }
+        }
+
+        return $resOptDetails;
+    }
+
+    /**
+     * @param \Magento\Catalog\Model\Product $product
+     * @param array $selectedTypes
+     * @return array
+     */
+    public function getOptionDetails(\Magento\Catalog\Model\Product $product, array $selectedTypes): array
```

Comment:
Looks like we can simplify the logic of the test preparation
Tests should be looks like documentation, now it's not readable
I will help you in this